Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -2767,6 +2767,14 @@ TextStream& operator<<(TextStream& ts, const WebProcessProxy& process)
     return ts;
 }
 
+void WebProcessProxy::validatePublicSuffix(const String& publicSuffix)
+{
+    bool isPublicSuffix = PublicSuffixStore::singleton().isPublicSuffix(publicSuffix);
+    if (!isPublicSuffix)
+        WEBPROCESSPROXY_RELEASE_LOG_ERROR(Network, "handleUnrecognizedPublicSuffix: Received an recognized public suffix %" PUBLIC_LOG_STRING, publicSuffix.utf8().data());
+    MESSAGE_CHECK(isPublicSuffix);
```

Comment:
What is the purpose of killing the child process here? A compromised WebProcess could easily NOT send this IPC..

Reply:
This is for collecting reports about unrecognized public suffix -- same as what the current assertion does.

Reply:
MESSAGE_CHECK() doesn't generate a crash log though.

Reply:
> same as what the current assertion does

Yes, I am questioning this assertion too. The UIProcess sends the WebProcess a Public Suffix, why is the WebProcess asserting that what it received is a public suffix? What's the purpose? Why can't the WebProcess trust the UIProcess? Also, clearly, the WebProcess is not even able to do the validation anyway.